lib/hashedUrls.js

```diff
@@ -37,19 +23,48 @@ function HostedPage(options) {
         // add secret value to hash arguments
         hashArgs += '&secret=' + params.secret;
 
-        var computedHash = urlencode(crypto.createHash('sha1').update(hashArgs).digest('base64')),
-            fullUrl = url + urlArgs + '&hash=' + computedHash;
+        var computedHash = urlencode(crypto.createHash('sha1').update(hashArgs).digest('base64'));
+        return url + urlArgs + '&hash=' + computedHash;
+    }
+
+    /**
+    * Create a hosted page URL
+    *
+    * @param {Object}      params
+    * @param {Function}    callback
+    */
+    function createHostedPageUrl(params, callback) {
+        var url = 'https://secure.rocketgate.com/hostedpage/servlet/HostedPagePurchase?';
 
-        callback(null, fullUrl);
+        if (params.dev) {
+            url = 'https://dev-secure.rocketgate.com/hostedpage/servlet/HostedPagePurchase?';
+            delete params.dev;
+        }
+        var fullUrlWithHash = computeHash(params, url);
+        return callback(null, fullUrlWithHash);
     }
 
+    /**
+    * Create a embedded fields form script url
+    *
+    * @param {Object}      params
+    * @param {Function}    callback
+    */
+    function createEmbeddedFieldsScript(params, callback) {
+        var url = 'https://secure.rocketgate.com/hostedpage/EmbeddedFields.jsp?';
 
+        if (params.dev) {
+            url = 'https://dev-secure.rocketgate.com/hostedpage/EmbeddedFields.jsp?';
+            delete params.dev;
+        }
+        var fullUrlWithHash = computeHash(params, url);
+        return callback(null, fullUrlWithHash);
+    }
 
-    publicAPI = {
-        createHostedPageUrl: createHostedPageUrl
+    return {
+        createHostedPageUrl: createHostedPageUrl,
+        createEmbeddedFieldsScript: createEmbeddedFieldsScript
     };
-
-    return publicAPI;
 }
 
-var exports = module.exports = HostedPage();
+module.exports = hashedUrls();
```

Comment:
What's the purpose of calling a function here? Surely we could just export directly?

Reply:
yeah, of course there are alot of bad practices in this sdk... i didn't want to do a lot of refactoring and overwhelm the RG guys, when i am doing an upstream PR.... but i agree, i can make this change too